apps/shipments/management/commands/replicate_shipments.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import sys
+import time
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from shipchain_common.exceptions import RPCError
+
+from apps.eth.models import EthAction
+from apps.eth.rpc import EventRPCClient
+from apps.jobs.models import AsyncJob, Message
+from apps.jobs.tasks import AsyncTask
+from apps.shipments.models import Shipment
+
+logger = logging.getLogger('transmission')
+logger.setLevel(settings.LOG_LEVEL)
+
+
+class Command(BaseCommand):
+    help = 'Replicate shipment related transactions to the sidechain.'
+    rpc_client = None
+    successfull_shipments = []
+    unsuccessfull_shipments = []
+    async_job = None
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--shipment_id',
+            type=str,
+            default=None,
+            help='Replicate specific shipment instead of all of them.',
+        )
+
+    def _replicate_shipment(self, shipment):
+        self.successfull_shipments.append(shipment.id)
+        for async_job in AsyncJob.objects.filter(shipment=shipment):
+            try:
+                eth_action = EthAction.objects.filter(async_job=async_job).first()
+                if eth_action:
+                    logger.info(f'Deleting Eth Action: {eth_action.transaction_hash}')
+                    eth_action.delete()
+                AsyncTask(async_job.id).rerun()
+                self.async_job = async_job
+
+            # pylint:disable=broad-except
+            except Exception as exc:
+                if shipment.id in self.successfull_shipments:
+                    self.successfull_shipments.remove(shipment.id)
+                    self.unsuccessfull_shipments.append(shipment.id)
+                logger.error(exc)
+
+    def _unsubscribe(self):
+        self.rpc_client = EventRPCClient()
+        try:
+            self.rpc_client.unsubscribe("LOAD", settings.LOAD_VERSION)
+        except RPCError as exc:
+            if 'Error: EventSubscription not found' not in exc.detail:
+                raise exc
+
+    def _resubscribe(self):
+        if not self.async_job:
+            logger.warning('Unable to find last blockheight to subscribe to. '
+                           'Need to manually subscribe to latest blockheight')
+            return
+
+        message = Message.objects.filter(async_job=self.async_job).last()
+        if "blockNumber" not in message.body:
+            time.sleep(15)
+            message = Message.objects.filter(async_job=self.async_job).last()
+        if "blockNumber" in message.body:
+            self.rpc_client.subscribe("LOAD", settings.LOAD_VERSION, last_block=message.body["blockNumber"])
+        else:
+            logger.warning('Unable to find last blockheight to subscribe to. '
+                           'Need to manually subscribe to latest blockheight')
+
+    def handle(self, *args, **options):
```

Comment:
Do we know the maximum runtime of a Django management command? Will it timeout after some time? If we run this via SSH, will it time out? Is there any sort of progress indicator provided via CLI output? What if this command takes days?

Reply:
The timing of this is based on the rerun times and the amount of shipments on txm. We can test locally by creating a couple 100 or so shipments via the creator, and then running the command. Could give us some indication on the time required for larger amounts of shipments.

Reply:
Still need more testing on this, and make sure that progress is displayed via CLI somehow. Maybe query for the total number of jobs to be run at the beginning and show a (count / total) indicator next to every log line.